app/src/main/java/com/example/swugger/EditEventDialogFragment.java

```diff
@@ -0,0 +1,313 @@
+            mTimeText.setTextColor(res.getColor(R.color.colorPrimary));
+        } else {
+            mTimeIcon.setColorFilter(getResources().getColor(R.color.black));
+            mTimeText.setTextColor(getResources().getColor(R.color.gray));
+        }
+    }
+
+    /** This method should not be called when Event.MAX_NUM_REMINDERS is reached because the button should be invisible.*/
+    @Override
+    public void onPositiveClickAddReminder(Integer daysBefore, Integer hoursBefore, Integer minutesBefore) {
+        long daysMilliseconds = daysBefore.longValue() * millisecondsPerDay;      // if days is super huge (not likely), watch for overflows
+        long hoursMilliseconds = hoursBefore.longValue() * millisecondsPerHour;
+        long minutesMilliseconds = minutesBefore.longValue() * millisecondsPerMinute;
+        long totalMilliseconds = daysMilliseconds + hoursMilliseconds + minutesMilliseconds;
+
+        // Get a Calendar representation of the date given by the original event
+        Calendar eventDate = Calendar.getInstance();
+        eventDate.set(mEvent.getYear(), mEvent.getMonth(), mEvent.getDay(), mEvent.getHour(), mEvent.getMinute());
+        // Calculate the date of the reminder
+        Calendar reminderDate = Calendar.getInstance();
+        long reminderDateInMilliseconds = eventDate.getTimeInMillis() - totalMilliseconds;
+        // eventDate should ALWAYS be >= reminderDate
+        if (reminderDateInMilliseconds < 0) {
+            throw new ArithmeticException("The reminder date should never be in the future.");
+        } else {
+            reminderDate.setTimeInMillis(eventDate.getTimeInMillis() - totalMilliseconds);
+        }
+
+        // Create separate Reminder objects in case we need to modify them later
+        newRemindersList.add(new Reminder(mEvent.getId(), reminderDateInMilliseconds, daysBefore, hoursBefore, minutesBefore));
+        displayedRemindersList.add(new Reminder(mEvent.getId(), reminderDateInMilliseconds, daysBefore, hoursBefore, minutesBefore));
+
+        // TODO: notify reminders adapter of change
+        mRemindersAdapter.notifyDataSetChanged();
+
+        // TODO: make the add reminder button invisible
+        mAddRemindersBtn.setVisibility(View.INVISIBLE);
+    }
+
+    @Override
+    public void onNegativeClickAddReminder() {
+        // Do nothing.
+    }
+
+    @Override
+    public void onDeleteReminder(Reminder reminder) {
+        // If the reminder has an Id, this means it's been added to the database and therefore must have its alarm canceled.
+        if (reminder instanceof ReminderWithId) {
+            remindersToDeleteList.add((ReminderWithId) reminder);
+        }
+        displayedRemindersList.remove(reminder);
+
+        mRemindersAdapter.notifyDataSetChanged();
+    }
+
+    /** Implements the EditEventsDialogListener so that any Activity
+     *  that implements it can retrieve information from this Dialog. */
+    public interface EditEventsDialogListener {
```

Comment:
Maybe take this out as its own interface file.

Reply:
i could when i need it as a more general purpose interface, but rn this is the only class that sends stuff to this kind of listener and will basically be the only class from what i have in mind